### DISCUSSION TITLE
[@types/chrome] runtime sendMessage typings are too strict

### DISCUSSION THREAD

**Discussion by tomasdev** [DefinitelyTyped/DefinitelyTyped#55765] [Issues with a @types package]
Added in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54890/files

There are not enough tests to cover the usage of `sendMessage`. There's a huge difference doing
```
export function sendMessage(message: any, responseCallback?: (response: any) => void): void;
```
versus doing
```
export function sendMessage<M = any, R = any>(message: M, responseCallback?: (response: R) => void): void;
```

For example, the former allows and the later doesn't the following:

```
export type ErrorCallback = (error: string) => void;

export type BooleanCallback = (result: boolean) => void;

export type Callback = ErrorCallback|BooleanCallback;

function sendExample(callback: Callback) {
  chrome.runtime.sendMessage('example', callback);
}
```

**Should it be reverted?**

The original PR quote:
> This change is backwards-compatiable because existing code will just use default values of any which will match prior behavior.

That statement is false. fn<T = any>(arg: T) is very different than fn(arg: any).

[a comment by an automated account is not shown]

**Comment by bershanskiy** [comment]
Thanks for posting this here. Did this error pop up during package upgrade? Which version are you upgrading from, if you don't mind me asking?

Removing existing templated declaration might break existing users who upgraded and changed declarations. Instead, I propose adding the old declaration next to the new one. Then when the new declaration/type inference backfires, TypeScript checker simply skips it and uses the old one with `any` types.

**Comment by tomasdev** [comment]
@bershankskiy I think the `fn(a: any)` syntax is backwards compatible with `fn<T = any>(a: T)` or at least couldn't reproduce a breakage.

I'm updating from a9fc1bf862e5eae9448146cffb106afacb304e6e (the version that was live on July 1st)
